migrations/sql/V2020.06.25.1104__AddVersion.sql

```diff
@@ -0,0 +1,15 @@
+CREATE TABLE version
+(
+    id SERIAL PRIMARY KEY,
+    user_id TEXT references users (id),
+    name TEXT NOT NULL,
+    note TEXT
+);
+
+ALTER TABLE user_session
```

Comment:
since this is no longer a session, can we rename the table accordingly (or probably just create a new table which can live alongside the old one for now so as not to be a breaking change)? presumably this new table won't need user_id either, which the existing user_session table has.

Reply:
another thought - will we need to retain the user_session table for non logged in users?

Reply:
I was holding off on renaming the table until we'd settled on terminology but can make a new table called snapshot or version_snapshot instead since it sounds like that's what we're going with for now. 
If we create a new table rather than repurposing the old one we'll have to fix up the FKs from session_file. Which I suppose should now be snapshot_file anyway. 
Guest users get a snapshot which isn't linked to a version. I guess that doesn't really need to have user_id in it. 

OK, so I can make the migration:
create new table version_snapshot with id and all the new columns, but no user id
create new table snapshot_file, with FK to version_snapshot
copy content from  user_session into version_snapshot
copy content from session_file into snapshot_file
(if we even want to retain content, maybe that's not so important since there's not much you can do with old sessions anyway - you just need what's in the file table to rehydrate a new session from an old one?)

Does that sound about right?

Reply:
Talked myself out of copying content..

Reply:
Agree with strategy of just creating new tables and no need to copy old content over :+1: